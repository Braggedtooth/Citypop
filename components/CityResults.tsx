import React from 'react'
import { View } from 'react-native'
import { Text, Card, useTheme } from '@rneui/themed'
import { CityResponse } from '../types/Response'
import { CityResultsProps } from '../types/Components'

const CityResults = ({ results }: CityResultsProps) => {
  const { theme } = useTheme()

  return (
    <Card containerStyle={{ borderRadius: 10, backgroundColor: theme.colors.searchBg }}>
      <Card.Title style={{ fontSize: 20 }} testID="title">
        Welcome to {results.name}
      </Card.Title>
      <View
        style={{
          alignItems: 'center',
          justifyContent: 'center',
          backgroundColor: theme.colors.black,
          padding: 10,
          borderRadius: 5
        }}
      >
        <Text
          testID="population"
          style={{
            fontSize: 40,
            color: theme.colors.success,
            fontWeight: 'bold'
          }}
        >
          {results.population.toLocaleString()}
        </Text>
      </View>
      <Text style={{ fontSize: 15, textAlign: 'center', marginTop: 5, color: theme.colors.greyOutline }}>
        People live here
      </Text>
    </Card>
  )
}

export default CityResults
