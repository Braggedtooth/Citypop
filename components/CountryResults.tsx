import { View } from 'react-native'
import React from 'react'
import { CountryResponse } from '../types/Response'
import { Text, useTheme } from '@rneui/themed'
import { useNavigation } from '@react-navigation/native'
import { homeScreenProps } from '../types/Navigator'

type CountryResultsProps = {
  results: CountryResponse
}
const CountryResults = ({ results }: CountryResultsProps) => {
  const { theme } = useTheme()
  const navigation = useNavigation<homeScreenProps>()
  const go = () => {
    navigation.navigate('Details', { name: results.name, population: results.population })
  }
  return (
    <View style={{ width: '100%' }}>
      <Text
        style={{
          color: theme.colors.white,
          backgroundColor: theme.colors.black,
          marginVertical: 5,
          padding: 15,
          marginHorizontal: 50,
          borderRadius: 5
        }}
        onPress={go}
      >
        {results.name}, {results.countryName}
      </Text>
    </View>
  )
}

export default CountryResults
