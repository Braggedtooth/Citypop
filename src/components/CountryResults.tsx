import { View } from 'react-native'
import React from 'react'
import { CountryResponse } from '../types/Response'
import { Text, useTheme } from '@rneui/themed'
import { useNavigation } from '@react-navigation/native'
import { homeScreenProps } from '../types/Navigator'
import { CountryResultsProps } from '../types/Components'
import useNavigate from '../utils/useNavigate'

const CountryResults = ({ results, details }: CountryResultsProps) => {
  const { theme } = useTheme()

  return (
    <View style={{ width: '100%' }}>
      <Text
        testID="countryInfo"
        style={{
          color: theme.colors.white,
          backgroundColor: theme.colors.black,
          marginVertical: 5,
          padding: 15,
          marginHorizontal: 50,
          borderRadius: 5
        }}
        onPress={() => details({ results })}
      >
        {`${results.name}, ${results.countryName}`}
      </Text>
    </View>
  )
}

export default CountryResults
