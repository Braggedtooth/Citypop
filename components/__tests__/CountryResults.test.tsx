import { fireEvent, render } from '@testing-library/react-native'
import { NavigationContainer } from '@react-navigation/native'
import React from 'react'
import CityResults from '../CityResults'
import CountryResults from '../CountryResults'

describe('Test City Results', () => {
  let props = { name: '', population: 0, countryName: '' }
  beforeEach(() => {
    props = {
      name: 'Test Country',
      population: 3000,
      countryName: 'Holland'
    }
  })
  const renderResults = () => render(<CountryResults results={props} details={() => true} />)

  it('Renders Correctly', () => {
    const component = renderResults().toJSON()
    expect(component).toMatchSnapshot()
  })
  it('Should render props', () => {
    const { getByTestId } = renderResults()
    const countryInfo = getByTestId('countryInfo')
    expect(countryInfo.props.children).toBe(`${props.name}, ${props.countryName}`)
  })
  it('Should be clickable', () => {
    const { getByTestId: country } = renderResults()
    const countryInfo = country('countryInfo')
    const press = fireEvent(countryInfo, 'press')
    expect(press).toBeTruthy()
  })
})
