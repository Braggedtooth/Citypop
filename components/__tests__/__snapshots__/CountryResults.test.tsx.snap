// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test City Results Renders Correctly 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <Text
    accessibilityRole="text"
    onPress={[Function]}
    style={
      Object {
        "backgroundColor": "#242424",
        "borderRadius": 5,
        "color": "#ffffff",
        "marginHorizontal": 50,
        "marginVertical": 5,
        "padding": 15,
      }
    }
    testID="countryInfo"
  >
    Test Country, Holland
  </Text>
</View>
`;
